### PR TITLE
Comet 038

### PR DESCRIPTION
**Pull request by p4u**

### CODE REVIEW

File: vochain/app.go
Comment (p4u) [inline vochain/app.go:1]: This is the relevant file for the review @jordipainan @altergui

File: vochain/app.go [hunk 1]
```diff
@@ -75,10 +74,18 @@ type BaseApplication struct {
 	circuitConfigTag    string
 	dataDir             string
 	genesisInfo         *tmtypes.GenesisDoc
+
+	// testMockBlockStore is used for testing purposes only
+	testMockBlockStore *testutil.MockBlockStore
```
Comment (jordipainan) [inline vochain/app.go:79]: seems strange to declare this here (just for testing purposes), but I am assuming it cannot be declared anywhere else.

File: vochain/app.go [hunk 2]
```diff
@@ -75,10 +74,18 @@ type BaseApplication struct {
...
 }
 
-// Ensure that BaseApplication implements abcitypes.Application.
-var _ abcitypes.Application = (*BaseApplication)(nil)
+// DeliverTxResponse is the response returned by DeliverTx after executing the transaction.
+type DeliverTxResponse struct {
```
Comment (jordipainan) [inline vochain/app.go:83]: Is not possible anymore to import this from comet types like we do in checktx ?

File: vochain/app.go [hunk 3]
```diff
@@ -224,84 +215,145 @@ func (app *BaseApplication) InitChain(req abcitypes.RequestInitChain) abcitypes.
...
+	for i, tx := range req.Txs {
+		resp := app.deliverTx(tx)
+		if resp.Code != 0 {
+			log.Warnw("deliverTx failed",
+				"code", resp.Code,
+				"data", string(resp.Data),
+				"info", resp.Info,
+				"log", resp.Log)
+		}
+		txResults[i] = &abcitypes.ExecTxResult{
```
Comment (jordipainan) [inline vochain/app.go:308]: should we include all txs even if the response code of the deliverTx is != 0 ?

File: vochain/app_benchmark_test.go [hunk 4]
```diff
@@ -126,20 +124,25 @@ func benchCheckTx(b *testing.B, app *BaseApplication, voters []*models.SignedTx)
...
-			detxresp = app.DeliverTx(detx)
+			detxresp := app.deliverTx(txBytes)
 			if detxresp.Code != 0 {
 				b.Fatalf(fmt.Sprintf("deliverTX failed: %s", detxresp.Data))
 			}
 		}
 		i++
 		if i%100 == 0 {
-			app.Commit()
+			_, err = app.Commit(context.TODO(), nil)
```
Comment (jordipainan) [inline vochain/app_benchmark_test.go:138]: if _, err := ...

File: vochain/app_benchmark_test.go [hunk 5]
```diff
@@ -126,20 +124,25 @@ func benchCheckTx(b *testing.B, app *BaseApplication, voters []*models.SignedTx)
...
+			if err != nil {
+				b.Fatal(err)
+			}
 		}
 	}
-	app.Commit()
+	_, err = app.Commit(context.TODO(), nil)
```
Comment (jordipainan) [inline vochain/app_benchmark_test.go:144]: if _, err := ...

File: vochain/apptest.go [hunk 6]
```diff
@@ -29,12 +29,15 @@ func TestBaseApplication(tb testing.TB) *BaseApplication {
 	if err != nil {
 		tb.Fatal(err)
 	}
-	app.InitChain(abcitypes.RequestInitChain{
+	_, err = app.InitChain(context.TODO(), &abcitypes.RequestInitChain{
```
Comment (jordipainan) [inline vochain/apptest.go:32]: if _, err := ...

File: vochain/apptest.go [hunk 7]
```diff
@@ -86,34 +89,23 @@ func (app *BaseApplication) SetTestingMethods() {
...
 // AdvanceTestBlock commits the current state, ends the current block and starts a new one.
 // Advances the block height and timestamp.
 func (app *BaseApplication) AdvanceTestBlock() {
-	app.Commit()
-	endingHeight := int64(app.Height())
-	app.EndBlock(abcitypes.RequestEndBlock{Height: endingHeight})
-
-	// The next block begins a second later.
-	nextHeight := endingHeight + 1
+	_, err := app.Commit(context.TODO(), nil)
```
Comment (jordipainan) [inline vochain/apptest.go:100]: if _, err := ...

File: vochain/genesis/genesis.go [hunk 8]
```diff
@@ -40,8 +40,8 @@ var Genesis = map[string]VochainGenesis{
 }
 
 var devGenesis = GenesisDoc{
-	GenesisTime: time.Date(2023, time.June, 5, 10, 0, 0, 0, time.UTC),
-	ChainID:     "vocdoni-dev-8",
+	GenesisTime: time.Date(2023, time.July, 5, 10, 0, 0, 0, time.UTC),
+	ChainID:     "vocdoni-dev-9",
```
Comment (jordipainan) [inline vochain/genesis/genesis.go:44]: Please next time comment on the PR that the updates are breaking

File: vochain/proof_erc20_test.go [hunk 9]
```diff
@@ -139,7 +138,10 @@ func testEthSendVotes(t *testing.T, s testStorageProof,
 			t.Fatalf("deliverTx success, but expected result is fail")
 		}
 	}
-	app.Commit()
+	_, err = app.Commit(context.TODO(), nil)
```
Comment (jordipainan) [inline vochain/proof_erc20_test.go:141]: if _, err := ...

File: vochain/proof_minime_test.go [hunk 10]
```diff
@@ -146,20 +143,24 @@ func testMinimeSendVotes(t *testing.T, s ethstorageproof.StorageProof, addr comm
...
 			t.Fatalf(fmt.Sprintf("deliverTx failed: %s", detxresp.Data))
 		}
 	} else {
 		if !expectedResult {
-			t.Fatalf("deliverTx success, but expected result is fail")
+			t.Fatalf("deliverTx uccess, but expected result is fail")
 		}
 	}
-	app.Commit()
+	_, err = app.Commit(context.TODO(), nil)
```
Comment (jordipainan) [inline vochain/proof_minime_test.go:160]: if _, err := ...

**Review by jordipainan — CHANGES_REQUESTED** [review]
added some comments

File: vochain/app.go (hunk 2 above)
Comment (p4u) [reply vochain/app.go:83]: its not

File: vochain/app.go (hunk 3 above)
Comment (p4u) [reply vochain/app.go:308]: this is just the result of the execution. So, you need to include a result even if the code is different to 0

**Pull request merged**